link/link_linux.go

```diff
@@ -0,0 +1,231 @@
+//go:build linux
+// +build linux
+
+package link
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"unsafe"
+
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+const (
+	netBasePath   = "/sys/class/net/"
+	netUEventPath = "/uevent"
+	netTypePath   = "/type"
+	netFlagsPath  = "/flags"
+
+	netUEventIfIndexPrefix    = "IFINDEX="
+	netUEventDevTypePrefix    = "DEVTYPE="
+	netUEventDevTypeVLAN      = "vlan"
+	netUEventIfIndexPrefixLen = len(netUEventIfIndexPrefix)
+	netUEventDevTypePrefixLen = len(netUEventDevTypePrefix)
+)
+
+var (
+	// ErrIndexOutOfBounds denotes the (unlikely) case of an invalid index being outside the range of an int
+	ErrIndexOutOfBounds = errors.New("interface index out of bounds")
+)
+
+// HostLinks returns all (or selected) host interfaces
```

Comment:
Super nit: you could document that this function will fail if any of the links cannot be created.